packages/caravan-health/src/waste.ts

```diff
@@ -0,0 +1,198 @@
+          on this transaction's change output plus the fees that will need to be paid 
+          to spend it later.
+      - Excess:
+          The amount by which we exceed our selection target when creating a changeless transaction, 
+          mutually exclusive with cost of change. It is extra fees paid if we don't make a change output 
+          and instead add the difference to the fees.
+      - Input Amount :
+          Sum of amount for each coin in input of the transaction
+      - Spend Amount :
+          Exact amount wanted to be spent in the transaction.
+
+    Calculation :
+      spend waste amount = consolidation factor + cost of transaction
+      spend waste amount = weight (fee rate - estimatedLongTermFeeRate) + change + excess
+
+    Observation :
+      Depending on the fee rate in the long term, the consolidation factor can either be positive or negative.		
+		    fee rate (current) < estimatedLongTermFeeRate (long-term fee rate)  –-> Consolidate now (-ve)
+		    fee rate (current) > estimatedLongTermFeeRate (long-term fee rate)  –-> Wait for later when fee rate go low (+ve) 
+
+  */
+  spendWasteAmount(
+    weight: number, // Estimated weight of the transaction
+    feeRate: number, // Current fee rate for the transaction
+    inputAmountSum: number, // Sum of amount for each coin in input of the transaction
+    spendAmount: number, // Exact amount wanted to be spent in the transaction
+    estimatedLongTermFeeRate: number, // Long term estimated fee-rate
+  ): number {
+    let costOfTx: number = Math.abs(spendAmount - inputAmountSum);
+    return weight * (feeRate - estimatedLongTermFeeRate) + costOfTx;
+  }
+
+  /*
+    Name : calculateDustLimits
+    Definition : 
+      "Dust" is defined in terms of dustRelayFee,
+      which has units satoshis-per-kilobyte.
+      If you'd pay more in fees than the value of the output
+      to spend something, then we consider it dust.
+      A typical spendable non-segwit txout is 34 bytes big, and will
+      need a CTxIn of at least 148 bytes to spend:
+      so dust is a spendable txout less than
+      182*dustRelayFee/1000 (in satoshis).
+      546 satoshis at the default rate of 3000 sat/kB.
+      A typical spendable segwit txout is 31 bytes big, and will
+      need a CTxIn of at least 67 bytes to spend:
+      so dust is a spendable txout less than
+      98*dustRelayFee/1000 (in satoshis).
+      294 satoshis at the default rate of 3000 sat/kB.
+
+    Calculation :
+      lowerLimit - Below which the UTXO will actually behave as a dust output.
+      upperLimit - Above which the UTXO will be safe and economical to spend.
+      riskMultiplier - A factor that helps to determine the upper limit.
+
+      lowerLimit = txWeight(default=182(34+148)) * feeRate (sats/vByte)
+      upperLimit = lowerLimit * riskMultiplier
+
+  */
+  calculateDustLimits(
+    feeRate: number,
+    // A typical spendable non-segwit txout is 34 bytes big
+    txWeight: number = 34,
```

Comment:
This should be done by passing in script type. txWeight is the wrong param/name for this. you're measuring the size of an input/witness to determine dust "factor" not the weight of a tx which you can't know when you're evaluating an individual utxo.